File: backend/models/statistics.go
package models

// Statistics represents collection statistics
type Statistics struct {
	TotalBlurays         int            `json:"total_blurays"`
	TotalMovies          int            `json:"total_movies"`
	TotalSeries          int            `json:"total_series"`
	TotalSeasons         int            `json:"total_seasons"`
	TotalEpisodes        int            `json:"total_episodes"`
	TotalSpent           float64        `json:"total_spent"`
	AveragePrice         float64        `json:"average_price"`
	PhysicalVolumeLiters float64        `json:"physical_volume_liters"`
	PhysicalStorageGB    float64        `json:"physical_storage_gb"`
	TotalRuntimeMinutes  int            `json:"total_runtime_minutes"`
	MostExpensive        *BlurayStats   `json:"most_expensive"`
	OldestBluray         *BlurayStats   `json:"oldest_bluray"`
	NewestBluray         *BlurayStats   `json:"newest_bluray"`
	GenreDistribution    map[string]int `json:"genre_distribution"`
	TagDistribution      map[string]int `json:"tag_distribution"`
	YearDistribution     map[int]int    `json:"year_distribution"`
	AverageRating        float64        `json:"average_rating"`
	TopRated             []BlurayStats  `json:"top_rated"`
}

type SimplifiedStatistics struct {
	TotalBlurays int `json:"total_blurays"`
	TotalMovies  int `json:"total_movies"`
	TotalSeries  int `json:"total_series"`
	TotalSeasons int `json:"total_seasons"`
}

// BlurayStats is a simplified bluray info for statistics
type BlurayStats struct {
	ID            string  `json:"id"`
	Title         string  `json:"title"`
	Type          string  `json:"type"`
	PurchasePrice float64 `json:"purchase_price,omitempty"`
	ReleaseYear   int     `json:"release_year,omitempty"`
	Rating        float64 `json:"rating,omitempty"`
}

File: frontend/src/types/statistics.ts
export interface BlurayStats {
  id: string;
  title: string;
  type: string;
  purchase_price?: number;
  release_year?: number;
  rating?: number;
}

export interface Statistics {
  total_blurays: number;
  total_movies: number;
  total_series: number;
  total_seasons: number;
  total_episodes: number;
  total_spent: number;
  average_price: number;
  physical_volume_liters: number;
  physical_storage_gb: number;
  total_runtime_minutes: number;
  most_expensive: BlurayStats | null;
  oldest_bluray: BlurayStats | null;
  newest_bluray: BlurayStats | null;
  genre_distribution: Record<string, number>;
  tag_distribution: Record<string, number>;
  average_rating: number;
  top_rated: BlurayStats[];
}

export interface SimplifiedStatistics {
  total_blurays: number;
  total_movies: number;
  total_series: number;
  total_seasons: number;
}
